upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -586,16 +586,29 @@ fi
 # to be removed from the system as part of an upgrade.
 function undeploy() {
   # Check if the chart exists by running helm status
-  # If the chart is missing (rc==1), return success.
-  helm status "$@" || return 0
-  # Remove the chart completely without keeping history.
-  helm uninstall "$@"
+  if helm status "$@"; then
```

Comment:
We should make sure it is actually deployed, it could be in other states
```suggestion
  if [[ $(helm status -o json "$@" 2>/dev/null | jq -r .info.status) == "deployed" ]]; then
```

Reply:
Here's my test:
```
pit:~ # cat dab.sh
#!/bin/bash 

function undeploy() {
  echo "Testing" "$@"
  # Check if the chart exists by running helm status
  if [[ $(helm status -o json "$@" 2>/dev/null | jq -r .info.status) == "deployed" ]]; then
    # Remove the chart completely without keeping history.
    echo helm uninstall "$@"
    return $?
  else
    return 0
  fi
}

# Undeploy Istio charts if they exist
# cray-istio-operator and cray-istio-deploy are removed with upgrade to Istio 1.26.0
undeploy -n istio-system cray-istio-operator
undeploy -n istio-system cray-istio-deploy
undeploy -n istio-system cray-istio-base

pit:~ # ./dab.sh
Testing -n istio-system cray-istio-operator
Testing -n istio-system cray-istio-deploy
Testing -n istio-system cray-istio-base
helm uninstall -n istio-system cray-istio-base
pit:~ # 
```